### PR TITLE
[MB-17008] Add left zero padding to LOA L1 segment values to meet 6 char validation

### PR DESCRIPTION
**Pull request by duncan-truss**
## Summary

We received feedback from DFAS about the EDI 858 LOA segments we were sending that the LOA L1 segment should always be six characters in length.  While some of the LOAs in our dataset have six character values for the `LoaInstlAcntgActID` others can be as small as three digit numerals (stored as string types).

This PR uses the format string to ensure we're left padding with zeroes to exactly 6 characters of length everytime unless we get a null value from the database.  In the case of a null value we will omit the L1 segment from the EDI like other fields instead of say sending "000000".  There is already a test that nil values get omitted so I didn't add a test for that.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

The easiest way to validate this PR is using the new test harness data:

1. Go to the test harness data: http://officelocal:3000/testharness/list
2. Click **MakeMoveReadyForEDI**
3. Search for and retain the payment request number it should look like: `"payment_request_number": "6841-8898-1"`
4. Build the generate-payment-request-edi cmd with `make bin/generate-payment-request-edi` to ensure you've compiled the latest changes.
5. Generate an EDI locally for the retained payment request number `bin/generate-payment-request-edi --payment-request-number <payment_request_number>`
6. Inspect the FA2 lines of the EDI, they should all contain the same data so verifying the first section is fine. You should see that the L1 value is now "000102" instead of "102", which is the value in the lines_of_accounting database table.
 
### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/about/supported-browsers) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/about/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots

Screenshot 2023-09-21 at 6 16 50 PM

### CODE REVIEW

File: pkg/services/invoice/ghc_payment_request_invoice_generator.go [hunk 1]
```diff
@@ -762,6 +762,11 @@ func (g ghcPaymentRequestInvoiceGenerator) createLongLoaSegments(appCtx appconte
 		concatDate = &fiscalYearStr
 	}
 
+	var accountingInstallationNumber *string
+	if loa.LoaInstlAcntgActID != nil {
+		zeroPaddedInstlAcntgActID := fmt.Sprintf("%06s", *loa.LoaInstlAcntgActID)
```
Comment (melissathai) [inline pkg/services/invoice/ghc_payment_request_invoice_generator.go:767]: non-blocking nit: for context for future devs, can a comment be added here stating that the DFAS requires this to be exactly 6 characters long?

**Review by melissathai — APPROVED** [review]
Just one non-blocking comment, otherwise looks reasonable to me.

I was wondering if this requirement should be enforced at the database level (it's currently a `varchar(6)`), but then decided against it since it's only used in one place in the codebase, and wouldn't be flexible if the character requirements change in the future. Curious on others' thoughts though.

**Comment by kctruss** [comment]
> since it's only used in one place in the codebase, and wouldn't be flexible if the character requirements change in the future

Agreed, I'm on the side of the DB reflecting accurate data and formatting only being done for the output.

**Review by kctruss — APPROVED** [review]
Thanks for tackling this @duncan-truss

**Comment by duncan-truss** [comment]
> Just one non-blocking comment, otherwise looks reasonable to me.
> 
> I was wondering if this requirement should be enforced at the database level (it's currently a `varchar(6)`), but then decided against it since it's only used in one place in the codebase, and wouldn't be flexible if the character requirements change in the future. Curious on others' thoughts though.

Thanks for reviewing @melissathai. Yea that discussion came up a week or so back during refinement about possibly backfilling the data.  It felt like extra effort that just made things a bit more inflexible.  For context too, there is a future 3rd party TGET integration that is going to supplant some of our data loading that we have had to do manually so far.

We tried to model some reasonable defaults in the database columns like you saw with a max 6 chars but it's unclear if those will hold going forward.  The best documentation we have is linked on the [Jira story](https://dp3.atlassian.net/browse/MB-17008) (TRDM File Layout - FA2 Values highlighted.pdf) that shows L1 is a mandatory field and that the min/max are both length 6.

I'll put a comment in the code too to make it clearer.

**Pull request merged**